js/client/modules/@arangodb/clusterstats.js

```diff
@@ -0,0 +1,115 @@
+/* jshint strict: false, sub: true */
+/* global print, arango */
+'use strict';
+
+// //////////////////////////////////////////////////////////////////////////////
+// / DISCLAIMER
+// /
+// / Copyright 2016 ArangoDB GmbH, Cologne, Germany
+// / Copyright 2014 triagens GmbH, Cologne, Germany
+// /
+// / Licensed under the Apache License, Version 2.0 (the "License")
+// / you may not use this file except in compliance with the License.
+// / You may obtain a copy of the License at
+// /
+// /     http://www.apache.org/licenses/LICENSE-2.0
+// /
+// / Unless required by applicable law or agreed to in writing, software
+// / distributed under the License is distributed on an "AS IS" BASIS,
+// / WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// / See the License for the specific language governing permissions and
+// / limitations under the License.
+// /
+// / Copyright holder is ArangoDB GmbH, Cologne, Germany
+// /
+// / @author Wilfried Goesgens
+// //////////////////////////////////////////////////////////////////////////////
+
+const internal = require('internal');
+const fs = require('fs');
+
+const download = internal.download;
+const time = internal.time;
+const sleep = internal.sleep;
+
+const pu = require('@arangodb/process-utils');
+
+const instanceInfo = JSON.parse(internal.env.INSTANCEINFO);
+const options = JSON.parse(internal.env.OPTIONS);
+const outfn = fs.join(instanceInfo.rootDir, 'stats.jsonl');
+const opts = Object.assign(pu.makeAuthorizationHeaders(options),
+                           { method: 'GET' });
+
+while(true) {
+  const before = time();
```

Comment:
hm, this one should measure one set of measurements - so I think its ok to have it outside.

Reply:
So it's used here: https://github.com/arangodb/arangodb/pull/9647/files#diff-7cd10cf13a409743cc8cee2bf4f5824cR104, and the check is to see if the iteration took more than 1000ms. The first iteration will succeed (probably), then we sleep for a second, then we check again _without updating_ `before`, and will inevitably fail. So I'm not sure exactly how you want the different `delta` and `before` values to work, but something's off here.

Reply:
it can have a look at the individual times of each server afterwards. If one (or some) of the requests are slow, this is the overal indicator somethings not good.

Reply:
*gna* had been looking at an older version, which didn't have this scope issue...